src/librustdoc/html/render/mod.rs

```diff
@@ -2403,7 +2405,7 @@ fn collect_paths_for_type(first_ty: clean::Type, cache: &Cache) -> Vec<String> {
             }
             clean::Type::QPath { self_type, trait_, .. } => {
                 work.push_back(*self_type);
-                work.push_back(*trait_);
+                process_path(trait_.res.def_id());
```

Comment:
This changes the order the paths are processed - previously `self_type` was processed first, now `trait_` is. Does that change the behavior?

Reply:
It looks like it will changes the order of the `Vec<String>` returned by the function this code is in (`collect_paths_for_type`). The order seems to be arbitrary though.

Reply:
:+1: looks like this is only used in https://github.com/rust-lang/rust/blob/5f744f33f60f1961e756adf30606a08076350182/src/librustdoc/html/render/mod.rs#L1632, which is order independent.

Reply:
I think it's also used when generating `implementors.js` files, but I think the order shouldn't matter there either.